mitiq/shadows/shadows.py

```diff
@@ -0,0 +1,199 @@
+    channel :math:`\mathcal{M}=\sum_b f_b\Pi_b`.
+
+    Args:
+        k_calibration: Number of groups of "median of means" used to solve for
+            Pauli fidelity.
+        locality: The locality of the operator, whose expectation value is
+            going to be estimated by the classical shadow. e.g. if operator is
+            Ising model Hamiltonian with nearist neighbour interaction, then
+            locality = 2.
+        zero_state_shadow_outcomes: The output of function
+            :func:`shadow_quantum_processing` of zero calibrate state.
+        qubits: The qubits to measure, needs to specify when the
+            `zero_state_shadow_outcomes` is None.
+        executor: The function to use to do quantum measurement, must be same
+            as executor in `shadow_quantum_processing`. Needs to specify when
+            the `zero_state_shadow_outcomes` is None.
+        num_total_measurements_calibration: Number of shots per group of
+            "median of means" used for calibration. Needs to specify when
+            the `zero_state_shadow_outcomes` is None.
+
+    Returns:
+        A dictionary containing the calibration outcomes.
+    """
+    if zero_state_shadow_outcomes is None:
+        if qubits is None:
+            raise ValueError("qubits should not be None.")
+        if executor is None:
+            raise ValueError("executor should not be None.")
+        if num_total_measurements_calibration is None:
+            raise ValueError(
+                "num_total_measurements_calibration should not be None."
+            )
+
+        # calibration circuit is of same qubit number with original circuit
+        zero_circuit = cirq.Circuit()
+        # perform random Pauli measurement one the calibration circuit
+        calibration_measurement_outcomes = random_pauli_measurement(
+            zero_circuit,
+            n_total_measurements=num_total_measurements_calibration,
+            executor=executor,
+            qubits=qubits,
+        )
+    else:
+        calibration_measurement_outcomes = zero_state_shadow_outcomes
+    # get the median of means estimation of Pauli fidelities
+    return get_pauli_fidelities(
+        calibration_measurement_outcomes, k_calibration, locality=locality
+    )
+
+
+def shadow_quantum_processing(
+    circuit: cirq.Circuit,
+    executor: Callable[[cirq.Circuit], MeasurementResult],
+    num_total_measurements_shadow: int,
+    random_seed: Optional[int] = None,
+    qubits: Optional[List[cirq.Qid]] = None,
+) -> Tuple[List[str], List[str]]:
+    r"""
+    This function returns the bit strings and Pauli strings corresponding to
+    the executor measurement outcomes for a given circuit, rotated by unitaries
+    randomly sampled from a fixed unitary ensemble :math:`\mathcal{G}`.
+
+    In the current implementation, the unitaries are sampled from the local
+    Clifford group for :math:`n` qubits, i.e.,
+    :math:`\mathcal{G} = \mathrm{Cl}_2^n`.
+
+    Args:
```

Comment:
```suggestion
    :math:`\mathcal{G} = \mathrm{Cl}_2^n`.
    In practice, the output of this functions provides the raw experimental
    data necessary to perform the classical shadows protocol.
    
    Args:
```

Reply:
Re-iterating Andrea's comment which was marked "outdated" after later changes.

Reply:
Thanks! I have checked this through!